### PR TITLE
Separate SchemaRegion Memory Implementation

### PR DESCRIPTION
**Pull request by MarcosZyk**
### Motivation
It is confusing to mix memory mode and schema file mode in one schemaRegion implementation, and it is hard to maintain schema file mode without affect memory mode.

The implementation of schema file mode already have a little effect on the performance of memory mode, due to try-finally and the Pin/UnPin and memory control mechanism. 

### Modification
Separate SchemaRegionMemoryImpl and SchemaRegionSchemaFileImpl, both implement ISchemaRegion.
Separate MTreeBelowSGMemoryImpl and MTreeBelowSGCachedImpl, both implement IMTreeBelowSG.

### CODE REVIEW

File: cluster/src/main/java/org/apache/iotdb/cluster/log/manage/PartitionedSnapshotLogManager.java [hunk 1]
```diff
@@ -130,7 +130,7 @@ void collectTimeseriesSchemas(List<Integer> requiredSlots) {
       }
       Collection<TimeseriesSchema> schemas =
           slotTimeseries.computeIfAbsent(slot, s -> new HashSet<>());
-      IoTDB.schemaProcessor.collectTimeseriesSchema(sgPath, schemas);
+      //      IoTDB.schemaProcessor.collectTimeseriesSchema(sgPath, schemas);
```
Comment (SilverNarcissus) [inline cluster/src/main/java/org/apache/iotdb/cluster/log/manage/PartitionedSnapshotLogManager.java:133]: Remove this comment

File: cluster/src/main/java/org/apache/iotdb/cluster/metadata/MetaPuller.java [hunk 2]
```diff
@@ -149,7 +148,7 @@ public void pullMeasurementSchemas(
       }
       int preSize = results.size();
       for (PartialPath prefixPath : prefixPaths) {
-        IoTDB.schemaProcessor.collectMeasurementSchema(prefixPath, results);
+        //        IoTDB.schemaProcessor.collectMeasurementSchema(prefixPath, results);
```
Comment (SilverNarcissus) [inline cluster/src/main/java/org/apache/iotdb/cluster/metadata/MetaPuller.java:151]: Remove this comment

File: cluster/src/main/java/org/apache/iotdb/cluster/query/LocalQueryExecutor.java [hunk 3]
```diff
@@ -485,8 +485,8 @@ private void collectSeries(List<String> prefixPaths, List<IMeasurementSchema> me
             .computeIfAbsent(slotPreviousHolderMap.get(slot), s -> new ArrayList<>())
             .add(new PartialPath(prefixPath));
       } else {
-        getCSchemaProcessor()
-            .collectMeasurementSchema(new PartialPath(prefixPath), measurementSchemas);
+        //        getCSchemaProcessor()
```
Comment (SilverNarcissus) [inline cluster/src/main/java/org/apache/iotdb/cluster/query/LocalQueryExecutor.java:488]: Remove this comment

File: cluster/src/test/java/org/apache/iotdb/cluster/server/member/MetaGroupMemberTest.java [hunk 4]
```diff
@@ -332,7 +332,8 @@ private PullSchemaResp mockedPullTimeSeriesSchema(PullSchemaRequest request) {
     try {
       for (String prefixPath : prefixPaths) {
         if (!prefixPath.equals(TestUtils.getTestSeries(10, 0))) {
-          IoTDB.schemaProcessor.collectMeasurementSchema(new PartialPath(prefixPath), schemas);
+          //          IoTDB.schemaProcessor.collectMeasurementSchema(new PartialPath(prefixPath),
```
Comment (SilverNarcissus) [inline cluster/src/test/java/org/apache/iotdb/cluster/server/member/MetaGroupMemberTest.java:335]: Remove this comment

**Review by SilverNarcissus — APPROVED** [review]
Brilliant refactoring, there are only minor issues, please have a look~

File: cluster/src/main/java/org/apache/iotdb/cluster/log/manage/PartitionedSnapshotLogManager.java (hunk 1 above)
Comment (MarcosZyk) [reply cluster/src/main/java/org/apache/iotdb/cluster/log/manage/PartitionedSnapshotLogManager.java:133]: The whole ```Cluster``` module will be removed after release of new cluster.

**Pull request merged**